batch/batch/front_end/front_end.py

```diff
@@ -1504,34 +1801,76 @@ async def _get_batch(app, batch_id):
   job_groups_n_jobs_in_complete_states.n_failed,
   job_groups_n_jobs_in_complete_states.n_cancelled,
   cost_t.*
-FROM batches
```

Comment:
Can you post the EXPLAIN of an example of this query? I'm just a bit curious to know what the join order looks like.

Reply:
```
mysql> EXPLAIN SELECT batches.*,
    ->   job_groups_cancelled.id IS NOT NULL AS cancelled,
    ->   job_groups_n_jobs_in_complete_states.n_completed,
    ->   job_groups_n_jobs_in_complete_states.n_succeeded,
    ->   job_groups_n_jobs_in_complete_states.n_failed,
    ->   job_groups_n_jobs_in_complete_states.n_cancelled,
    ->   cost_t.*
    -> FROM job_groups
    -> LEFT JOIN batches ON batches.id = job_groups.batch_id
    -> LEFT JOIN job_groups_n_jobs_in_complete_states
    ->        ON job_groups.batch_id = job_groups_n_jobs_in_complete_states.id AND job_groups.job_group_id = job_groups_n_jobs_in_complete_states.job_group_id
    -> LEFT JOIN job_groups_cancelled
    ->        ON job_groups.batch_id = job_groups_cancelled.id AND job_groups.job_group_id = job_groups_cancelled.job_group_id
    -> LEFT JOIN LATERAL (
    ->   SELECT COALESCE(SUM(`usage` * rate), 0) AS cost, JSON_OBJECTAGG(resources.resource, COALESCE(`usage` * rate, 0)) AS cost_breakdown
    ->   FROM (
    ->     SELECT resource_id, CAST(COALESCE(SUM(`usage`), 0) AS SIGNED) AS `usage`
    ->     FROM aggregated_job_group_resources_v3
    ->     WHERE job_groups.batch_id = aggregated_job_group_resources_v3.batch_id AND job_groups.job_group_id = aggregated_job_group_resources_v3.job_group_id
    ->     GROUP BY resource_id
    ->   ) AS usage_t
    ->   LEFT JOIN resources ON usage_t.resource_id = resources.resource_id
    -> ) AS cost_t ON TRUE
    -> WHERE job_groups.batch_id = 5 AND job_groups.job_group_id = 0 AND NOT deleted;
+----+-------------------+--------------------------------------+------------+--------+-------------------------------------------------------------------------------------+-------------+---------+---------------------------+------+----------+-----------------------------------------+
| id | select_type       | table                                | partitions | type   | possible_keys                                                                       | key         | key_len | ref                       | rows | filtered | Extra                                   |
+----+-------------------+--------------------------------------+------------+--------+-------------------------------------------------------------------------------------+-------------+---------+---------------------------+------+----------+-----------------------------------------+
|  1 | PRIMARY           | job_groups                           | NULL       | const  | PRIMARY,job_groups_state_callback,job_groups_time_created,job_groups_time_completed | PRIMARY     | 12      | const,const               |    1 |   100.00 | Using index; Rematerialize (<derived2>) |
|  1 | PRIMARY           | batches                              | NULL       | const  | PRIMARY,batches_deleted                                                             | PRIMARY     | 8       | const                     |    1 |   100.00 | NULL                                    |
|  1 | PRIMARY           | job_groups_n_jobs_in_complete_states | NULL       | const  | PRIMARY                                                                             | PRIMARY     | 8       | const                     |    1 |   100.00 | NULL                                    |
|  1 | PRIMARY           | job_groups_cancelled                 | NULL       | const  | PRIMARY                                                                             | PRIMARY     | 8       | const                     |    0 |     0.00 | unique row not found                    |
|  1 | PRIMARY           | <derived2>                           | NULL       | ALL    | NULL                                                                                | NULL        | NULL    | NULL                      |    2 |   100.00 | Using where                             |
|  2 | DEPENDENT DERIVED | <derived3>                           | NULL       | ALL    | NULL                                                                                | NULL        | NULL    | NULL                      |    2 |   100.00 | NULL                                    |
|  2 | DEPENDENT DERIVED | resources                            | NULL       | eq_ref | resource_id                                                                         | resource_id | 4       | usage_t.resource_id       |    1 |   100.00 | NULL                                    |
|  3 | DEPENDENT DERIVED | aggregated_job_group_resources_v3    | NULL       | ref    | PRIMARY,resource_id                                                                 | PRIMARY     | 8       | batch.job_groups.batch_id |   21 |    10.00 | Using where; Using temporary            |
+----+-------------------+--------------------------------------+------------+--------+-------------------------------------------------------------------------------------+-------------+---------+---------------------------+------+----------+-----------------------------------------+
8 rows in set, 3 warnings (0.01 sec)
```